### DISCUSSION TITLE
maplibre + mbtiles offline maps

### DISCUSSION THREAD

**Discussion by sjchristi** [rnmapbox/maps#2113] [General]
Hi all,

In my app, I'm trying to support both online and offline rendering of maps.  I've got a bunch of `.mbtiles` files generated for geographic locations roughly the size of a small country / or US State.  Before users go offline, they can download mbtiles files for different regions to use when they are offline.

I'm using v8.5.0 of the library (I'm happy to retest on main if it would make it easier).  I've got a simple example setup that allows you to download an mbtile file for West Virginia, and then allows you to switch between online and offline modes.

I have a style file bundled with the app.  If you hit the 'offline' button, the styleJSON for the map is setup like this:

```js
    const newMapStyle = JSON.parse(JSON.stringify(mapStyle));
    newMapStyle.sources.openmaptiles = {
        type: 'vector',
        // This works only on iOS; on Android it causes tile errors:
        // tiles: [`mbtiles://file://${path}`],
        // This works on Android and iOS:
        url: `mbtiles://file://${path}`,
      };
      return JSON.stringify(newMapStyle);
```

As you can see, I can make it work on iOS if i use the `tiles:` line -- and I can only make it work on android or iOS if I use the `url:` line.

I'd prefer to use the `tiles:` syntax if possible because it looks like I could add a list of `.mbtiles` to the array in the case they've download multiple offline regions (not sure how I'd do that with the `url` syntax...)

Would anyone happen to know why this might be the case?  Maybe this is just a difference in the iOS / Android maplibre libraries and I should really be asking them?

I'm happy to share my simple example if it would help shed any light..

thanks,
Sam

**Comment by Cdingram** [comment]
Has anyone been able to make this work at all recently? I haven't been able to make either options work, both give me Mapbox error Map load failed: {begin=2977762474096, source-id=omv, message=Failed to load source: Expected URL scheme 'http' or 'https' but was 'mbtiles', type=source}.

I'm using the example file at the bottom of this page: https://docs.mapbox.com/help/glossary/mbtiles/

**Reply by CarlosAlbertoTI** [reply]
No, both options doesn't work for me.
